app/controllers/admin/results_controller.rb

```diff
@@ -42,10 +45,14 @@ def start_posting
       @updated_competitions = Competition.pending_posting.where(posting_user: nil).where(id: params[:competition_ids])
       return render json: { error: "No competitions to lock." } if @updated_competitions.empty?
 
-      json = { error: "Something went wrong." }
-      json = { message: "Competitions successfully locked, go on posting!" } if @updated_competitions.update(posting_user: current_user)
+      ActiveRecord::Base.transaction do
+        @updated_competitions.update(posting_user: current_user)
+        ticket_competition_result_ids = @updated_competitions.joins(:tickets_competition_result).pluck('tickets_competition_result.id')
+        TicketsCompetitionResult.where(id: ticket_competition_result_ids)
+                                .update_all(status: TicketsCompetitionResult.statuses[:warnings_verification])
```

Comment:
Here's one more thing to try: If you define a `belongs_to :competition` directly on `TicketsCompetitionResult`, then doing `TicketsCompetitionResult.where(competition: @updated_competitions).update_all(...)` might work

Reply:
I am not sure, since we're not doing these composite model approaches very often, but please at least try it out to see whether it works or not

Reply:
Yes, that worked. Great solution, thank you.